controllers/vrg_status_pvcs.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"github.com/go-logr/logr"
+	ramen "github.com/ramendr/ramen/api/v1alpha1"
+)
+
+// findProtectedPVC returns the &VRG.Status.ProtectedPVC[x] for the given pvcName
+func (v *VRGInstance) findProtectedPVC(pvcName string) *ramen.ProtectedPVC {
+	return FindProtectedPVC(v.instance, pvcName)
+}
+
+func FindProtectedPVC(vrg *ramen.VolumeReplicationGroup, pvcName string) *ramen.ProtectedPVC {
+	protectedPvc, _ := FindProtectedPvcAndIndex(vrg, pvcName)
+
+	return protectedPvc
+}
+
+func (v *VRGInstance) pvcStatusDeleteIfPresent(pvcName string, log logr.Logger) {
+	pvcStatus, i := FindProtectedPvcAndIndex(v.instance, pvcName)
+	if pvcStatus == nil {
+		log.Info("PVC status absent already")
+
+		return
+	}
+
+	log.Info("PVC status delete", "index", i)
+	v.instance.Status.ProtectedPVCs = sliceUnorderedElementDelete(v.instance.Status.ProtectedPVCs, i)
+}
+
+func sliceUnorderedElementDelete[T any](s []T, i int) []T {
+	s[i] = s[len(s)-1]
+
+	return s[:len(s)-1]
+}
+
+func FindProtectedPvcAndIndex(vrg *ramen.VolumeReplicationGroup, pvcName string) (*ramen.ProtectedPVC, int) {
+	for index := range vrg.Status.ProtectedPVCs {
+		protectedPVC := &vrg.Status.ProtectedPVCs[index]
+		if protectedPVC.Name == pvcName {
+			return protectedPVC, index
+		}
+	}
+
+	return nil, len(vrg.Status.ProtectedPVCs)
```

Comment:
The pattern followed by most languages is that if an element is not present in the array, the returned index would be -1. If you return the length of the array, that would be misleading for someone reusing this function.

Reply:
C++ returns the "past-the-end" reference
https://en.cppreference.com/w/cpp/container/unordered_map/find
> Return value
Iterator to an element with key equivalent to key. If no such element is found, past-the-end (see [end()](https://en.cppreference.com/w/cpp/container/unordered_map/end)) iterator is returned.

Reply:
That's an iterator, which is similar to a pointer.

Reply:
If you return an index, you have to return a common invalid index.  Look at this Java IndexOf as an example
http://www.java2s.com/example/java/collection-framework/counterpart-to-javautilcollectionindexof-for-array.html